### PR TITLE
fix: localize timeout in test_cancel_replication_immediately test

### PR DESCRIPTION
**Pull request by romange**

### CODE REVIEW

File: src/server/replica.cc [hunk 1]
```diff
@@ -81,7 +81,8 @@ error_code Replica::Start(ConnectionContext* cntx) {
   ProactorBase* mythread = ProactorBase::me();
   CHECK(mythread);
 
-  RETURN_ON_ERR(cntx_.SwitchErrorHandler(absl::bind_front(&Replica::DefaultErrorHandler, this)));
+  RETURN_ON_ERR(
```
Comment (romange) [inline src/server/replica.cc:84]: removed redundant "Replica::DefaultErrorHandler"

File: tests/dragonfly/replication_test.py [hunk 2]
```diff
@@ -423,7 +425,8 @@ async def test_cancel_replication_immediately(df_local_factory, df_seeder_factor
     df_local_factory.start_all([replica, master])
 
     seeder = df_seeder_factory.create(port=master.port)
-    c_replica = aioredis.Redis(port=replica.port)
+    c_replica = aioredis.Redis(port=replica.port, socket_timeout=20)
```
Comment (romange) [inline tests/dragonfly/replication_test.py:428]: added timeout to socket operations

File: tests/dragonfly/replication_test.py [hunk 3]
```diff
@@ -437,14 +440,20 @@ async def replicate():
             return False
 
     for i in range(COMMANDS_TO_ISSUE):
-        replication_commands.append(replicate())
-    results = await asyncio.gather(*replication_commands)
-    num_successes = sum(results)
+        replication_commands.append(asyncio.create_task(replicate()))
+    num_successes = 0
+    for result in asyncio.as_completed(replication_commands, timeout=30):
```
Comment (romange) [inline tests/dragonfly/replication_test.py:445]: added timeout on waiting for tasks

File: src/server/replica.cc (hunk 1 above)
Comment (dranikpg) [reply src/server/replica.cc:84]: Yeah, it probably appeared when it was refactored

**Review by dranikpg — APPROVED** [review]

**Comment by kostasrim** [comment]
@romange @dranikpg do not merge this, I can reproduce locally, I will take care of this soon + I got the logs now. It's fixable

**Comment by romange** [comment]
@kostasrim  I think the fix is orthogonal to how the test is written. Before this PR I had to guess where it deadlocks in a test. With this PR I know exactly what step deadlocks and that's what I mean by improving the quality of regression tests.

**Pull request merged**